teps/0066-dogfooding-tekton.md

```diff
@@ -0,0 +1,320 @@
+> that a company is confident about its products.
+
+We would like to capture the goals of the dogfooding effort in the Tekton community,
+as document guiding principles, existing achievements as well as future roadmap.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The dogfooding effort in Tekton has been going on since the early stages of the project,
+so why writing a TEP now?
+
+When the dogfooding work was started we did not have a TEP process in place, we relied on
+[shared Google documents](https://docs.google.com/document/d/1Tf3tdIBwkN5kxMuyYMo802F-DyOdsl_bgh-78a9prWQ/edit#)
+instead. This TEP captures the work done this far, its guiding principles, the future
+roadmap; it aims to raise awareness about this work in the community, and to make it
+easier for new developers to get started and contribute to it.
+
+### Goals
+
+Tekton is non-opinionated by design. It allows designing CI/CD pipelines and systems,
+but it's not prescriptive about how to do that. The dogfooding effort offers CI/CD
+services to the Tekton community through Tekton, and in doing so it must take design
+decisions on how to build such services through Tekton. Goals of this TEP are to:
+
+- Formalize the motivation for the dogfooding effort
+- Describe the design principles for CI/CD services based on Tekton
+- Identify the services implemented through Tekton
+- Identify areas were more work is needed and define a future roadmap
+
+### Non-Goals
+
+This TEP does not aim to design specific services based on Tekton, that work will be tracked
+in dedicated TEPs where needed.
+
+### Use Cases (optional)
+
+Use cases that the dogfooding effort implements are:
+
+- Experience Tekton from the operator, author and user point of view, which allows the
+  Tekton community to:
+  - Identify missing functional features and usability issues
+  - Identify operational problems and security concerns
+  - Discover bugs that can be better identified in a running system, such as upgrade issues,
+    a problems related to maintainability and scale
+- Offer CI/CD services to the Tekton community with minimal dependencies
+  to other products, and thus greater control on the outcome
+- Build a repository of examples that embody best practices for Tekton use to tap in
+
+## Requirements
+
+- Provide reliable CI/CD services to the Tekton community
+- Dogfood as many project projects as possible, from the most stable and mature to
+  the experimental ones
+- Provide a mean to experiment with alpha and experimental features and build a way
+  forward to stable
```

Comment:
i might add a requirement in here around maybe who would be providing these services and how easy it is for folks to contribute? maybe something like 'a low barrier to entry for "tekton owners" and/or contributors to contribute to dogfooding efforts'

Reply:
Sounds good - it should always be "easy" to contribute in the sense that there should be no unnecessary complexity and good documentation. In terms of policies for merging PRs, we have a single repo `plumbing` which contains resources / configurations for both production environments like `prow` and `dogfooding` as well as more experimental ones like `robocat`. Perhaps we could create different set of owners for different parts of the repo?

Reply:
done